src/GraphQL/Builders/FieldBuilder.cs

```diff
@@ -139,6 +141,25 @@ protected FieldBuilder(FieldType fieldType)
         public virtual FieldBuilder<TSourceType, TReturnType> ResolveAsync(Func<IResolveFieldContext<TSourceType>, Task<TReturnType?>> resolve)
             => Resolve(new FuncFieldResolver<TSourceType, TReturnType>(context => new ValueTask<TReturnType?>(resolve(context))));
 
+        /// <inheritdoc cref="Resolve(IFieldResolver)"/>
+        public virtual FieldBuilder<TSourceType, TReturnType> Resolve(Delegate? resolve)
```

Comment:
I'm concerned about this.  This means that any delegate might unintentionally compile:

```cs
    public DroidType(StarWarsData data)
    {
        Field<StringGraphType>("help")
            .Returns<bool>()
            .Resolve(Test);
    }

    protected virtual string Test(IResolveFieldContext<Droid> context)
    {
        return "help"; // expecting bool
    }
```

Without the `Delegate` overload, this would have been a compilation error requiring a `bool` return type.

Also note that my concern does not affect inline functions; apparently they cannot resolve to `Delegate`.

Thoughts:
- Is this a "feature" that you can just pass any method to `Resolve`?
- Should we verify the return type and/or arguments against those defined in the builder?
- Should we automatically add arguments that were not defined with `Argument()`?
- Can we infer the graph type of the field?  (the underlying code already has this ability)

For example:

```cs
    public DroidType(StarWarsData data)
    {
        Field("help").Resolve(Help);
    }

    private Character Help(Assistance assistanceNeeded)
    {
        if (assistanceNeeded == Assistance.Fighting)
            return data.Find("Chewie");
        if (assistanceNeeded == Assistance.Force)
            return data.Find("Luke");
    }
```

Taking this thought process a step further, I've always been dissatisfied that the field builders ask for the graph type of the field.  Why?  The expression syntax does not need it (`Field(x => x.Name)`).  If we are specifying the return type of the field, we should not need to specify the graph type of the field.

```cs
    public DroidType(StarWarsData data)
    {
        // we should be able to do this
        Field("help")
            .Returns<bool>()
            .ResolveAsync(async context => true);

        // or better yet, this
        Field<bool>("help")
            .ResolveAsync(async context => true);

        // but this is annoying and stupid (personal opinion)
        Field<BooleanGraphType>("help")
            .Returns<bool>()
            .ResolveAsync(async context => true);
    }
```

I think if we are refactoring/enhancing the field builders we should figure this out.

Reply:
To carry on my suggestion, I'd suggest:

```cs
// already have this
public virtual FieldBuilder<TSourceType, object> Field<TGraphType>(string name = "default")
    where TGraphType : IGraphType { }

// suggesting to add this
/// <summary>
/// Adds a new field to the complex graph type and returns a builder for this newly added field.
/// </summary>
public virtual FieldBuilder<TSourceType, TReturnType> Field<TReturnType>(string name, bool nullable = false)
{
    var type = typeof(TReturnType);

    try
    {
        if (type == null)
            type = typeof(TReturnType).GetGraphTypeFromType(nullable, this is IInputObjectGraphType ? TypeMappingMode.InputType : TypeMappingMode.OutputType);
    }
    catch (ArgumentOutOfRangeException exp)
    {
        throw new ArgumentException($"The GraphQL type for field '{Name ?? GetType().Name}.{name}' could not be derived implicitly from type '{typeof(TReturnType).Name}'.", exp);
    }

    var builder = CreateBuilder<TReturnType>(type)
        .Name(name);

    AddField(builder.FieldType);
    return builder;
}

// both are valid:
Field<BooleanGraphType>("hello");
Field<bool>("hello");
```

Reply:
Re-thinking the above sample with a delegate, and assuming the graph type, attributes and query arguments are defined from the delegate, then I'm not sure the field builder is appropriate anyway:

```cs
    public DroidType(StarWarsData data)
    {
        // maybe this doesn't make sense, since no graph type can be inferred until Resolve is called:
        Field("help").Resolve(Help);

        // perhaps this is better
        Field("help", GetHelp);  // if we assume that functions typically have internal names, or if it makes sense to have the field names all listed in the constructor, this might be good

        // and/or this, where the name is inferred
        Field(Help);  // still could use `[Name]` attribute to define the name, even with no name defined here
    }

    private Character Help(Assistance assistanceNeeded) { }
```

Note that with a delegate, NRT attributes can be read from the delegate, whereas with inline functions they cannot.

Reply:
Well, the things are simpler. At first attemt I named this method `ResolveDelegate`. And at some point of rewriting I decided to rename it into `Resolve` to align with other methods. I think the simple solution here - rename it to `ResolveDelegate` back to get rid of all potential problems and to not compilate the code. OK?

Reply:
I'm sure very few people use this method.

Reply:
Seems fine for this PR. Maybe consider further.

Reply:
Changed